apps/test/unit/templates/studioHomepages/SectionsTableTest.js

```diff
@@ -0,0 +1,164 @@
+      <td>
+        <div>
+          {i18n.section()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.course()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.teacher()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.sectionCode()}
+        </div>
+      </td>
+    );
+    // Students with teacher-managed accounts should not see a button to leave the section.
+    expect(wrapper.find('Button').exists()).to.be.false;
+  });
+
+  it('shows column headers for students who do not have teacher-managed accounts', () => {
+    const wrapper = shallow(
+      <SectionsTable
+        sections={joinedSections}
+        isRtl={false}
+        isTeacher={false}
+        canLeave={true}
+      />
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.section()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.course()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.teacher()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.sectionCode()}
+        </div>
+      </td>
+    );
+    // Students who do not have teacher-managed accounts should see a button to leave the section.
+    expect(wrapper.find('Button').exists()).to.be.true;
+  });
+
+  it('shows column headers for students teachers', () => {
+    const wrapper = shallow(
+      <SectionsTable
+        sections={joinedSections}
+        isRtl={false}
+        isTeacher={true}
+        canLeave={false}
+      />
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.section()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.course()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.students()}
+        </div>
+      </td>
+    );
+    expect(wrapper).to.containMatchingElement(
+      <td>
+        <div>
+          {i18n.sectionCode()}
+        </div>
+      </td>
+    );
+    // Teachers can't be students in their own sections, therefore do not see a button to leave the section.
+    expect(wrapper.find('Button').exists()).to.be.false;
+  });
+
+  it('renders a row for each joined section', () => {
+    const wrapper = shallow(
+      <SectionsTable
+        sections={joinedSections}
+        isRtl={false}
+        isTeacher={false}
+        canLeave={false}
+      />
+    );
+    const randomJoinedSectionIndex = Math.floor(Math.random() * 4);
```

Comment:
Is this a reasonable approach?  Testing each field in each row seemed like overkill, but in the long run this might be harder to debug since it's randomly spot checking.

Reply:
I'm not sure testing all four is overkill, and any time we use randomness there's a chance we're introducing something flaky into our tests.  I'd loop over the four instead.

```jsx
expect(wrapper.find('.row')).to.have.length(4);
joinedSections.forEach((section) => {
  expect(wrapper).to.containMatchingElement(
    <td>
      <div>
        {section.name}
      </div>
    </td>
  );
  expect(wrapper).to.containMatchingElement(
    <td>
      <a>
        {section.assignedTitle}
      </a>
    </td>
  );
  // etc...
});
```